cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Query.hs

```diff
@@ -244,6 +247,21 @@ hprop_cli_queries = integrationWorkspace "cli-queries" $ \tempAbsBasePath' -> H.
       pure ()
 
     )
+  where
+  patchGovStateOutput :: String -> Either String String
+  patchGovStateOutput output = do
+    eOutput <- eitherDecodeStrictText (T.pack output)
+    return $ T.unpack $ decodeUtf8 $ prettyPrintJSON $ patchGovStateJSON eOutput
+    where
+      patchGovStateJSON :: Aeson.Object -> Aeson.Object
```

Comment:
What's the explanation for this patch?

Reply:
There is non-determinism in `futurePParams`, it seems there is a point in the `epoch` when the chain is not sure about the value, so it says it is unknown

Reply:
In particular, sometimes it returns:
```json
    "futurePParams": {
        "tag": "NoPParamsUpdate"
    },
```
And others:
```json
    "futurePParams": {
        "contents": null,
        "tag": "PotentialPParamsUpdate"
    },
```

Reply:
Ok, with the new function (that waits for the `(6*k/f)`th slot) I haven't seen it return anything other than `NoPParamsUpdate`, and I tested it quite a few times. So the patch doesn't seem necessary anymore